langchain/document_loaders/trello.py

```diff
@@ -0,0 +1,126 @@
+        include_card_name (bool): Whether to include the name of the card in the document. Defaults to False.
+        include_comments (bool): Whether to include the comments on the card in the document. Defaults to False.
+        include_checklist (bool): Whether to include the checklist on the card in the document. Defaults to False.
+        card_filter (str, optional): Use "closed" / "open". Defaults to "all".
+        extra_metadata (tuple[str]): List of additional metadata fields to include as document metadata. Defaults to ["due_date", "labels", "list", "is_closed"].
+
+    """
+
+    def __init__(
+        self, 
+        api_key: str, 
+        api_token: str,
+        board_name: str,
+        include_card_name: bool = True,
+        include_comments: bool = True,
+        include_checklist: bool = True,
+        card_filter: str = "all",
+        extra_metadata: tuple[str, ...] = ("due_date", "labels", "list", "is_closed"),
+    ):
+        """Initialize Trello loader."""
+        self.api_key = api_key
+        self.api_token = api_token
+        self.board_name = board_name
+        self.include_card_name = include_card_name
+        self.include_comments = include_comments
+        self.include_checklist = include_checklist
+        self.extra_metadata = extra_metadata
+        self.card_filter = card_filter
+
+    def load(self) -> List[Document]:
+        """Loads all cards from the specified Trello board.
+        You can filter the cards, metadata and text included by using the optional parameters.
+
+         Returns:
+            A list of documents, one for each card in the board.
+        """
+
+        try:
+            from trello import TrelloClient  # type: ignore
+        except ImportError as ex:
+            raise ImportError(
+                "Could not import trello python package. "
+                "Please install it with `pip install py-trello`."
+            ) from ex
+        try:
+            from bs4 import BeautifulSoup  # type: ignore
+        except ImportError as ex:
+            raise ImportError(
+                "`beautifulsoup4` package not found, please run"
+                " `pip install beautifulsoup4`"
+            ) from ex
+
+        docs: List[Document] = []
+        client = TrelloClient(api_key=self.api_key, token=self.api_token)
+
+        # Find the board with the matching name
+        board = next((b for b in client.list_boards()
+                     if b.name == self.board_name), None)
+        if board:
+            board_id = board.id
+        else:
+            raise ValueError(f"Board `{self.board_name}` not found.")
+
+        # Create a dictionary with the list IDs as keys and the list names as values
+        list_dict = {list_item.id: list_item.name for list_item in board.list_lists()}
+            
+        # Get Cards on the board
+        board = client.get_board(board_id)
+        cards = board.get_cards(card_filter=self.card_filter)
+        for card in cards:
```

Comment:
board_get_cards():
`[
  {
    "id": "55550aca6952888df7975903",
    "name": "Camera",
    "description": "",
    "closed": false,
    "labels": [],
    "due_date": "",
    "checklists": [
      [
        {
          "name": "Zoom",
          "state": "complete"
        },
        {
          "name": "Follow players",
          "state": "complete"
        },
        {
          "name": "camera limit to stage size",
          "state": "complete"
        },
        {
          "name": "Post Processing effects",
          "state": "complete"
        },
        {
          "name": "Shitch to universal render pipeline",
          "state": "complete"
        }
      ]
    ],
    "comments": [
      {
        "data": {
          "test": "to follow group of players use Group Camera feature of cinemachine."
        }
      },
      {
        "data": {
          "test": "Use 'Impulse' Cinemachine feature for camera shake."
        }
      },
      {
        "data": {
          "test": "depth of field with custom shader."
        }
      }
    ]
  },
  {
    "id": "555579f6eaf4d9055d7aa53a",
    "name": "Custom Collision logic during movement",
    "description": "",
    "closed": false,
    "labels": [],
    "due_date": "",
    "checklists": [],
    "comments": []
  },
  {
    "id": "55555257cecdf6458f53c969",
    "name": "Explosion Object",
    "description": "",
    "closed": false,
    "labels": [],
    "due_date": "",
    "checklists": [
      [
        {
          "name": "Create explosion prefrab (variant of AnimationEffect)",
          "state": "complete"
        },
        {
          "name": "Add custom collider + particle emitter",
          "state": "complete"
        },
        {
          "name": "Add collision logic, destroy bullets and expulse players.",
          "state": "complete"
        },
        {
          "name": "Add diff sizes",
          "state": "complete"
        },
        {
          "name": "Diff emitter for big and small sub explosions. Other particles tweaks, opacity.",
          "state": "complete"
        },
        {
          "name": "Collider only enabled on first frame.",
          "state": "complete"
        },
        {
          "name": "Add additional small particles with bouncing properties.",
          "state": "complete"
        }
      ]
    ],
    "comments": []
  },
  {
    "id": "5555e8964c62a3677fb8b534",
    "name": "Unify .ase files",
    "description": "",
    "closed": false,
    "labels": [],
    "due_date": "",
    "checklists": [],
    "comments": []
  },
  {
    "id": "55557bee3223ec56db323000",
    "name": "Automated import into Sprite Sheets and animation clips.",
    "description": "We are using a fork. Reuse the parsing logic but implement diff layer processing, creating 1 clip per animation to allow programmatically clip swipe on running time.\nBased on naming convention on layers we create a folder structure matching each diff skin.",
    "closed": false,
    "labels": [],
    "due_date": "",
    "checklists": [],
    "comments": []
  },
  {
    "id": "5555b0729a68da84ede0c0f5",
    "name": "Dynamic Skin on runtime script",
    "description": "PlayerSkin class: does Auto-loading / Assigning dynamically animations clips to override animator controllers.",
    "closed": false,
    "labels": [],
    "due_date": "",
    "checklists": [],
    "comments": []
  },
  {
    "id": "5555da301c66de7286ab5d76",
    "name": "add a UI overlay to the GameModeSelect scene for networking messages",
    "description": "",
    "closed": false,
    "labels": [],
    "due_date": "",
    "checklists": [],
    "comments": []
  }
]`